analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.cs

```diff
@@ -63,16 +70,131 @@ public sealed class MemberInitializerRedundant : SonarDiagnosticAnalyzer
+                }
+
+                if (expression is MemberAccessExpressionSyntax memberAccess
+                    && memberAccess.Expression.IsKind(SyntaxKind.ThisExpression))
+                {
+                    identifier = memberAccess.Name as IdentifierNameSyntax;
+                }
+
+                if (expression is ConditionalAccessExpressionSyntax conditionalAccess
+                    && conditionalAccess.Expression.IsKind(SyntaxKind.ThisExpression))
+                {
+                    identifier = (conditionalAccess.WhenNotNull as MemberBindingExpressionSyntax)?.Name as IdentifierNameSyntax;
+                }
+
+                if (identifier == null)
+                {
+                    return false;
+                }
+
+                var assignedSymbol = semanticModel.GetSymbolInfo(identifier).Symbol;
+
+                return memberToCheck.Equals(assignedSymbol);
+            }
+
+            public bool TryGetReadWriteFromMemberAccess(ExpressionSyntax expression, out bool isRead)
+            {
+                isRead = false;
+
+                var parenthesized = expression.GetSelfOrTopParenthesizedExpression();
+
+                if (!IsMatchingMember(expression))
+                {
+                    return false;
+                }
+
+                if (IsOutArgument(parenthesized))
+                {
+                    isRead = false;
+                    return true;
+                }
+
+                if (IsReadAccess(parenthesized, this.semanticModel))
+                {
+                    isRead = true;
+                    return true;
+                }
+
+                return false;
+            }
+
+            private static ExpressionSyntax GetPossibleMemberAccessParent(IdentifierNameSyntax identifier)
+            {
+                if (identifier.Parent is MemberAccessExpressionSyntax memberAccess)
+                {
+                    return memberAccess;
+                }
+
+                if (identifier.Parent is MemberBindingExpressionSyntax memberBinding)
+                {
+                    return (ExpressionSyntax)memberBinding.Parent;
+                }
+
+                return identifier;
+            }
+
+            private static bool IsBeingAssigned(ExpressionSyntax expression) =>
+                expression.Parent is AssignmentExpressionSyntax assignment
+                && assignment.IsKind(SyntaxKind.SimpleAssignmentExpression)
+                && assignment.Left == expression;
+
+            private static bool IsOutArgument(ExpressionSyntax parenthesized) =>
+                parenthesized.Parent is ArgumentSyntax argument
+                && argument.RefOrOutKeyword.IsKind(SyntaxKind.OutKeyword);
+
+            private static bool IsReadAccess(ExpressionSyntax parenthesized, SemanticModel semanticModel) =>
+                !IsBeingAssigned(parenthesized)
+                && !parenthesized.IsInNameOfArgument(semanticModel);
+        }
+
```

Comment:
This `InitializationVerifierFactory` is a triple indirection - really impossible to read and understand.
And all we get is unified 20 lines of code inside `Report` method, but we need 50 complicated lines to orchestrate it.

Let's just remove `InitializationVerifierFactory` and `InitializationVerifier` completely. It will make the code flat and straightforward. There will be two different implemenations for the current `Report` method, something like `CheckStaticMembers` and `CheckInstanceMembers`